### PR TITLE
Add crowdstrike lookback option

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4511?filter=-1&jql=assignee%20%3D%20currentUser()%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20summary%20ASC%2C%20updated%20DESC)

## Description
Add lookback otption to crowdstrike fetch incidents
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -1978,25 +1982,54 @@ def get_mapping_fields_command() -> GetMappingFieldsResponse:
...
     if not last_fetch_time:
-        last_fetch_time, _ = parse_date_range(FETCH_TIME, date_format='%Y-%m-%dT%H:%M:%SZ')
+        last_fetch_time, _ = parse_date_range(FETCH_TIME, date_format=DATE_FORMAT)
     prev_fetch = last_fetch_time
     last_fetch_timestamp = int(parse(last_fetch_time).timestamp() * 1000)
     return last_fetch_time, offset, prev_fetch, last_fetch_timestamp
 
 
+def migrate_last_run():
+    last_run = demisto.getLastRun()
```
Comment (ilaner) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:1996]: add handle_last_run to check conditions for migration

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 2]
```diff
@@ -1978,25 +1982,54 @@ def get_mapping_fields_command() -> GetMappingFieldsResponse:
...
+    if not last_run:
+        last_run = [{}, {}]
+    current_fetch_info_detections: dict = last_run[0]
+    current_fetch_info_incidents: dict = last_run[1]
     fetch_incidents_or_detections = demisto.params().get('fetch_incidents_or_detections')
+    look_back = int(demisto.params().get('look_back', 0))
+    fetch_time = demisto.params().get('fetch_time')
+    fetch_limit = INCIDENTS_PER_FETCH
 
+    demisto.debug("CrowdstrikeFalconMsg: Starting fetch incidents")
```
Comment (ilaner) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:2025]: add configuration to fetch

Comment (ilaner) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:2025]: print lastrun in debug

File: Packs/Base/ReleaseNotes/1_31_33.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (yuvalbenshalom) [inline Packs/Base/ReleaseNotes/1_31_33.md:1]: remove the file changes.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -27,6 +27,7 @@
 from distutils.version import LooseVersion
 from threading import Lock
 from inspect import currentframe
+import pytz
```
Comment (yuvalbenshalom) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:30]: remove the file changes

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_9_10.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### CrowdStrike Falcon
+- This version requires **Base** pack version `1.31.33`
```
Comment (yuvalbenshalom) [inline Packs/CrowdStrikeFalcon/ReleaseNotes/1_9_10.md:4]:
```suggestion
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good. see my comments and please implement the changes in your comments.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**